go/vt/sqlparser/ast.go

```diff
@@ -2343,12 +2343,11 @@ type (
 		Expr         Expr
 	}
 
-	// TimestampFuncExpr represents the function and arguments for TIMESTAMP{ADD,DIFF} functions.
-	TimestampFuncExpr struct {
-		Name  string
+	// TimestampDiffExpr represents the function and arguments for TIMESTAMPDIFF functions.
+	TimestampDiffExpr struct {
```

Comment:
This now only has `TIMESTAMPDIFF` left, so make it specialized for that. It might be something we want to treat as an interval or not in the future, but not sure yet so leaving it here for now. Did change the parsing here as well though to match `TIMESTAMPADD` for the interval types it supports.